### DISCUSSION TITLE
Directx 12 or Vulkan in Osu Lazer

### DISCUSSION THREAD

**Discussion by TheRitrax** [ppy/osu#26383] [Ideas]
The vast majority of GPUs are better optimized for Directx 12 or Vulkan than the Directx 11 or OpenGL ones that osu lazer currently uses.

Let's say as an example, the GPU that I use from Intel (Arc a770) has better optimized performance and is more stable in Directx 12 or Vulkan, that is, it is better optimized in modern APIs.

In the case of Nvidia, no matter the API, the performance will always be stable, although not the highest possible.

And in the case of AMD, it is better optimized in Vulkan than in Directx of any version, because AMD helped in the development of Vulkan and has better performance and stability in its GPUs.

Although what is more likely to happen is that Vulkan is better optimized than Directx 12.

**Comment by peppy** [comment]
vulkan is planned for a future, but "optimised for direct x 12" means crap all. direct x is backwards compatible and the changes from 11 to 12 will not benefit us much.

**Reply by smoogipoo** [reply]
This is not correct. "Optimised for DirectX 12" in terms of _Intel ARC_ means that they use a D3D11-to-12 layer in the driver to get it to work. Doing native D3D12 would be beneficial for this particular GPU.

But I have tested on an Intel ARC A380 and it runs fine.
